File: Leetcode/Python/3151. Special Array I/solution.py
# https://leetcode.com/problems/special-array-i/description

class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        for idx in range(1, len(nums)):
            if nums[idx - 1] & 1 == nums[idx] & 1:
                return False

        return True

File: Leetcode/C++/3151. Special Array I/solution.cpp
// https://leetcode.com/problems/special-array-i/description

class Solution
{
public:
    bool isArraySpecial(vector<int> &nums)
    {
        for (int idx = 1; idx < nums.size(); idx++)
        {
            if ((nums[idx - 1] & 1) == (nums[idx] & 1))
            {
                return false;
            }
        }

        return true;
    }
};